src/homeblks/volume/volume.cpp

```diff
@@ -335,7 +335,96 @@ indx_tbl* Volume::recover_indx_tbl(btree_super_block& sb, btree_cp_sb& cp_info)
     return static_cast< indx_tbl* >(tbl);
 }
 
+static bool is_buf_zero(const uint8_t* buf, size_t size) {
+    // TODO: subsample the buffer to detect zero request instead of working on the whole buffer to achieve constant
+    //  processing time for large buffer size requests. Needs to investigate the performance impact of this change
+    //  in end2end testing.
+    static std::vector< uint8_t > read_buf(size, 0);
+    static const auto zero_crc = crc16_t10dif(init_crc_16, read_buf.data(), size);
+    const auto crc = crc16_t10dif(init_crc_16, buf, size);
```

Comment:
One suggestion is the size is 4k aligned, we can calculate 4k, 8k, 16k, 32k, 64k, 128k, 256k all zero crc on boot (initialize only once) and store them in hs_utils to access, so that it won't need to be calculated in every I/O. I think it could cover most of the use cases for all-zero crcs.

Reply:
done